### ISSUE TITLE
Don't run dexcount task if the inputs haven't changed

### ISSUE THREAD

**Issue by garyp** [KeepSafe/dexcount-gradle-plugin#266]
It'd be nice if dexcount supported [gradle's incremental build mode](https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:up_to_date_checks). I.e. the task should only run if the APK/AAR has changed. The input file to the task is already properly annotated with `@InputFile`, so at a minimum all that's needed is to also annotate `DexMethodCountTaskBase.summaryFile` with `@OutputFile`.

As a proof of concept, adding this to my project's `build.gradle` properly skips the dexcount task on subsequent runs:

```
tasks.withType(DexMethodCountTaskBase).configureEach {
    outputs.file({ summaryFile })
}
```

For a fully-correct implementation, you'll probably want to annotate all of the outputs (e.g. the `chartDir`) and annotate all of the config options too.

**Comment by benjamin-bader** [comment]
I _believe_ we once were fully incremental, and then decided against it.  Most of our users (as measured by github issues) seemed confused that they didn't see method counts on each and every build, and IMO it's hopeless to expect everyone to become intimately familiar with task inputs and outputs vis-a-vis Gradle incrementality.

All that to say, that was _also_ the origin of the `runOnEachPackage` config option, which I assume you're familiar with given how much work you've already done.  What's the behavior of the plugin with that flag set to false, and how does it differ from what you'd expect?

I'm guessing that it still counts more often than strictly necessary; if so, it's worth investigating whether we can annotate outputs properly and still preserve the current out-of-the-box behavior.

**Comment by benjamin-bader** [comment]
I want to say that this is resolved by today's release of version 1.0.0.  We've annotated basically every input to the task, so all that remains is to use the aforementioned `runOnEachPackage = false` configuration.
